File: src/main/java/com/temelyan/pomoapp/service/UserService.java
package com.temelyan.pomoapp.service;

import com.temelyan.pomoapp.model.User;
import com.temelyan.pomoapp.to.UserTo;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;

import java.time.LocalDateTime;

public interface UserService extends UserDetailsService {
    User update(UserTo userTo);

    User update(User user);

    User create(User user);

    @Override
    UserDetails loadUserByUsername(String username) throws UsernameNotFoundException;

    User loadByEmail(String email);

    User findUserByResetToken(String token);

    User getByIdWithPomosInDateRange(int userId, LocalDateTime from, LocalDateTime to);
}

File: src/main/java/com/temelyan/pomoapp/service/UserService.kt
package com.temelyan.pomoapp.service

import com.temelyan.pomoapp.AuthorizedUser
import com.temelyan.pomoapp.model.User
import com.temelyan.pomoapp.to.UserTo
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException

interface UserService : UserDetailsService {
    fun update(userTo: UserTo): User

    fun update(user: User): User

    fun create(user: User): User

    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(username: String): AuthorizedUser

    fun loadByEmail(email: String): User

    fun findUserByResetToken(token: String): User
}
